src/Neo/Ledger/Blockchain.cs

```diff
@@ -152,17 +154,32 @@ public Blockchain(NeoSystem system)
 
         private void OnImport(IEnumerable<Block> blocks, bool verify)
         {
+            var blocksToImport = blocks as Block[] ?? blocks.ToArray();
+            var hash = blocksToImport.FirstOrDefault()?.NextConsensus;
+
+            // Capture necessary context elements
+            var self = Self;
+            var verificationTasks = blocksToImport.Select(block => Task.Run(() =>
```

Comment:
Can we avoid this tasks when verify == false?

Reply:
Need to do `await Task.WhenAll(verificationTasks);` This is a synchronous process. Currently this will have the whole process out of sync

Reply:
if verify is false, we don't need to do it

Reply:
> Need to do `await Task.WhenAll(verificationTasks);` This is a synchronous process. Currently this will have the whole process out of sync. For example if one were to fail. This is would break the system.

I was trying to do an optimize importing, assuming it works, then fail the system if anything in the thread is wrong and show errors.